cookbook/controller/error_pages.rst

```diff
@@ -5,172 +5,154 @@
-    the router throws an exception (for instance, when the route does not
-    match), then using ``is_granted`` will throw a further exception. You
-    can use ``is_granted`` safely by saying ``{% if app.user and is_granted('...') %}``.
+    The exception pages shown in the development environment can be customized
+    in the same way as error pages. Create a new ``exception.html.twig`` template
+    for the standard HTML exception page or ``exception.json.twig`` for the JSON
+    exception page.
 
-.. tip::
+Avoiding Exceptions when Using Security Functions in Error Templates
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    If you're not familiar with Twig, don't worry. Twig is a simple,
-    powerful and optional templating engine that integrates with
-    Symfony. For more information about Twig see :doc:`/book/templating`.
+One of the common pitfalls when designing custom error pages is to use the
+``is_granted()`` function in the error template (or in any parent template
+inherited by the error template). If you do that, you'll see an exception thrown
+by Symfony.
 
-This works not only to replace the default templates, but also to add
-new ones.
+The cause of this problem is that routing is done before security. If a 404 error
+occurs, the security layer isn't loaded and thus, the ``is_granted()`` function
+is undefined. The solution is to add the following check before using this function:
 
-For instance, create an ``app/Resources/TwigBundle/views/Exception/error404.html.twig``
-template to display a special page for 404 (page not found) errors.
-Refer to the previous section for the order in which the
-``ExceptionController`` tries different template names.
+.. code-block:: twig
 
-.. tip::
+    {% if app.user and is_granted('...') %}
+        {# ... #}
+    {% endif %}
 
-    Often, the easiest way to customize an error page is to copy it from
-    the TwigBundle into ``app/Resources/TwigBundle/views/Exception`` and
-    then modify it.
+Testing Error Pages during Development
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. note::
+One of the biggest hurdles of testing how do custom error pages look in your
+application is the fact that Symfony ignores them in the development environment
+and displays instead the default exception pages.
 
-    The debug-friendly exception pages shown to the developer can even be
-    customized in the same way by creating templates such as
-    ``exception.html.twig`` for the standard HTML exception page or
-    ``exception.json.twig`` for the JSON exception page.
+You may be tempted to set the ``kernel.debug`` parameter to ``false`` to disable
+the debug mode in the development environment. However, this practice is not
+recommended because it will also stop Symfony from recompiling your Twig templates,
+among many other things.
+
+The recommended solution is to use a third-party bundle called `WebfactoryExceptionsBundle`_.
```

Comment:
Do not forgot to change this part for >=2.6 as the main functionality of WebfactoryExceptionsBundle has been integrated into Symfony core.